pkg/controller/rmdworkload/rmdworkload_controller_test.go

```diff
@@ -470,6 +495,1169 @@ func TestRmdWorkloadControllerReconcile(t *testing.T) {
 			t.Errorf("Failed: %v - Expected %v, got %v", tc.name, tc.expectedError, expectedError)
 		}
 		ts.Close()
+		ts2.Close()
+	}
+}
+
+func TestFindObseleteWorkloads(t *testing.T) {
+	tcases := []struct {
+		name                      string
+		rmdNodeStateList          *intelv1alpha1.RmdNodeStateList
+		request                   reconcile.Request
+		rmdWorkload               *intelv1alpha1.RmdWorkload
+		rmdPods                   *corev1.PodList
+		getWorkloadsResponse      map[string]([]rmdtypes.RDTWorkLoad)
+		expectedObseleteWorkloads map[string]string
+		expectedErr               bool
+	}{
+		{
+			name: "test case 1 - 1 obselete workload only",
+			rmdNodeStateList: &intelv1alpha1.RmdNodeStateList{
+				Items: []intelv1alpha1.RmdNodeState{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rmd-node-state-example-node.com",
+							Namespace: "default",
+						},
+						Spec: intelv1alpha1.RmdNodeStateSpec{
+							Node: "example-node.com",
+						},
+					},
+				},
+			},
+			request: reconcile.Request{
+				types.NamespacedName{
+					Namespace: "default",
+					Name:      "rmd-workload-1",
+				},
+			},
+			rmdWorkload: &intelv1alpha1.RmdWorkload{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "rmd-workload-1",
+					Namespace: "default",
+				},
+				Spec: intelv1alpha1.RmdWorkloadSpec{
+					Nodes: []string{"example-node.com"},
+				},
+			},
+			rmdPods: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rmd-example-node.com",
+							Namespace: "default",
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Ports: []corev1.ContainerPort{
+										{
+											ContainerPort: 8080,
+										},
+									},
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							PodIPs: []corev1.PodIP{
+								{
+									IP: "127.0.0.1",
+								},
+							},
+						},
+					},
+				},
+			},
+			getWorkloadsResponse: map[string]([]rmdtypes.RDTWorkLoad){
+				"127.0.0.1:8080": {
+					{
+						UUID: "rmd-workload-1",
+						ID:   "1",
+					},
+				},
+			},
+			expectedObseleteWorkloads: map[string]string{
+				"http://127.0.0.1:8080": "rmd-workload-1",
```

Comment:
Due to the change mentioned in above comment (replacing `workload.UUID` with `workload.ID`), a lot of these test cases in this function will now break. For instance, the expected result here will actually be `"http://127.0.0.1:8080": "1"`